### PR TITLE
Give integration tests unique temp folder names

### PR DESCRIPTION
**Pull request by DanTup**
This will help track down any that aren't cleaning up and also may help track down leaked flutter_tester processes (https://github.com/dart-lang/sdk/issues/35549).

### CODE REVIEW

**Review by gspencergoog — APPROVED** [review]

**Comment by DanTup** [comment]
I believe the bot failures were related to the flaky `flutter test` integration tests which are currently skipped (though have a PR open that I think fixes them). Rebased in this on the skipped version so hopefully will go green and I'll land this.

**Comment by DanTup** [comment]
Ugh :(

```
 [+ 30621] TimeoutException: Request to Dart VM Service timed out: _reloadSources({pause: false, rootLibUri: file:///C:/Users/ContainerAdministrator/AppData/Local/Temp/flutter_run_expression_evaluation_test.bbb07e6c-18dd-11e9-a94a-f16e5f480ef5c2ef7d86-18dd-11e9-a94a-f16e5f480ef5/flutter_run_expression_evaluation_test.bbb07e6c-18dd-11e9-a94a-f16e5f480ef5/lib/main.dart.incremental.dill, packagesUri: file:///C:/Users/ContainerAdministrator/AppData/Local/Temp/flutter_run_expression_evaluation_test.bbb07e6c-18d...
  [+ 30621] #0      VM.invokeRpcRaw (package:flutter_tools/src/vmservice.dart:834:9)
```

Certainly unrelated to these changes.

File: packages/flutter_tools/test/integration/test_utils.dart [hunk 1]
```diff
@@ -14,8 +14,8 @@ import '../src/common.dart';
 /// Creates a temporary directory but resolves any symlinks to return the real
 /// underlying path to avoid issues with breakpoints/hot reload.
 /// https://github.com/flutter/flutter/pull/21741
-Directory createResolvedTempDirectorySync() {
-  final Directory tempDir = fs.systemTempDirectory.createTempSync('flutter_expression_test.');
+Directory createResolvedTempDirectorySync(String prefix) {
+  final Directory tempDir = fs.systemTempDirectory.createTempSync(prefix);
```
Comment (Hixie) [inline packages/flutter_tools/test/integration/test_utils.dart]: maybe factor out the common parts: `'flutter_$prefix.'` ?

Comment (DanTup) [reply packages/flutter_tools/test/integration/test_utils.dart]: Done!

**Comment by DanTup** [comment]
I'm now less convinced the Windows failure is unrelated flake... Three times the Windows bot has failed with:

```
15:03 +541 ~5 -1: test\integration\expression_evaluation_test.dart: flutter run expression evaluation can evaluate trivial expressions in top level function [E]                                       
  Did not receive response to request "1".
  Received:
  [+    12] [{"event":"app.progress","params":{"appId":"dbb67c17-7098-450b-bf3f-1c500efef36b","id":"1","progressId":"hot.reload","message":"Initializing hot reload..."}}]
  [+   463] [{"event":"app.progress","params":{"appId":"dbb67c17-7098-450b-bf3f-1c500efef36b","id":"2","progressId":null,"message":"Syncing files to device Flutter test device..."}}]
  [+   555] [{"event":"app.progress","params":{"appId":"dbb67c17-7098-450b-bf3f-1c500efef36b","id":"2","progressId":null,"finished":true}}]
  [+ 30567] Unhandled exception:
  [+ 30567] TimeoutException: Request to Dart VM Service timed out: _reloadSources({pause: false, rootLibUri: file:///C:/Users/ContainerAdministrator/AppData/Local/Temp/flutter_run_expression_evaluation_test.dffb14c5-196c-11e9-a94a-ba12d7d213dce6fe5253-196c-11e9-a94a-ba12d7d213dc/flutter_run_expression_evaluation_test.dffb14c5-196c-11e9-a94a-ba12d7d213dc/lib/main.dart.incremental.dill, packagesUri: file:///C:/Users/ContainerAdministrator/AppData/Local/Temp/flutter_run_expression_evaluation_test.dffb14c5-196...
```

I don't know why this change would cause the reload to time out, but I'll do some testing on my own Windows machine today.

**Comment by DanTup** [comment]
Ok, I'm a bit stumped. I have narrowed the failure down to this specific change:

screenshot 2019-01-17 at 8 02 50 am

I ran just these tests on my fork without that change, and it passes:

https://cirrus-ci.com/task/4739105164361728

With that change, it fails:

https://cirrus-ci.com/task/5094808047583232

This is consistent with the results I had in this PR while changing them.

Part of me wants to just land it without that change (all the names are still unique) but part of me also wants to know how changing the name of a random temp folder can result in timeouts to the Dart VM service 🤕

**Pull request merged**